cachestore/composite/cache_store.go

```diff
@@ -0,0 +1,77 @@
+package composite
+
+import (
+	"context"
+	"github.com/merlinapp/datarepo-go"
+	"github.com/merlinapp/datarepo-go/drreflect"
+	"time"
+)
+
+type compositeCacheStore struct {
+	delegates []datarepo.CacheStore
+}
+
+// Creates a new composite CacheStore backed by the provider CacheStore implementations.
+//
+// Read operations will be delegated down to the provided caches in the order they're provided.
+// If a cache returns a result (key is found), then that result will be used and further caches will not be queried.
+//
+// Write operations will be propagates to all delegate caches
+func NewCompositeCacheStore(delegates ...datarepo.CacheStore) datarepo.CacheStore {
+	if len(delegates) == 0 {
+		panic("Can't create a composite cache store with no delegate caches")
+	}
+
+	store := compositeCacheStore{
+		delegates: delegates,
+	}
+	return &store
+}
+
+func (c *compositeCacheStore) Set(ctx context.Context, key string, value interface{}, expiration time.Duration) {
+	for _, cache := range c.delegates {
+		cache.Set(ctx, key, value, expiration)
+	}
+}
+
+func (c *compositeCacheStore) Get(ctx context.Context, key string, out interface{}) (bool, error) {
+	var err error
+	for _, cache := range c.delegates {
+		found, err2 := cache.Get(ctx, key, out)
```

Comment:
I think it's better to rename this variable to c_err

Reply:
Done, although I used `cErr` instead as the IDE didn't like `c_err` in Go :)